### ISSUE TITLE
`[WARNING] The following source file contents changed or missing since the latest analysis`

### ISSUE THREAD

**Issue by RobertoFioravanti** [Ericsson/codechecker#4021]
I don't think it's a bug.
I'm doing something wrong.

 

I'm trying to store a report, which has been generated by github workflow action,  using the command
`CodeChecker  store /path/to/report  -n first_run --url http://localhost:8001/Default --verbose debug`.

I get this warning and the Product page is empty
```
[WARNING][2023-09-26 06:47:11] {report-converter} [51] <140523418298176> - reports.py:51 dump_changed_files() - The following source file contents changed or missing since the latest analysis:
 - /home/runner/work/codechecker-analysis-action-test/codechecker-analysis-action-test/src/main.cpp
Please re-analyze your project to update the reports!
```

Codechecker version: 6.22.2
Codechecker is running in a docker container ( [docker-compose.sqlite](https://github.com/Ericsson/codechecker/blob/master/web/docker/services/docker-compose.sqlite.yml) )

The following file is the zipped report I'm trying to store
[report.zip](https://github.com/Ericsson/codechecker/files/12723819/report.zip)

**Comment by whisperity** [comment]
This warning appears due to a self-defence logic in CodeChecker. If the analysis at time point **A** produced warnings to a file, but the file has changed at later time **B**, when the `store` command uploads the results, it would upload the warnings as were created for version **A**, but the contents of the file from version **B**.

The problem is that the warnings contain information, such as line numbers, function names, macro expansions, etc., that are only valid for the version analysed and could get invalid due to subsequent changes.

You must not let anything (e.g., CI jobs running in parallel to the analysis that touch the same directories) change the contents of the files before `store` succeeds. Otherwise, this defence will be triggered.

**Comment by RobertoFioravanti** [comment]
Thank you. 
In my case the Default product is empty: codechecker is just installed.
I've forked [codechecker-analysis-action-test](https://github.com/whisperity/codechecker-analysis-action-test) and I've executed the analysis triggering the workflow: after the analysis files have not been changed. 
I changed the original workflow action in order to upload the report as a .zip file. Then I downloaded the report and I used it in the `store` command. So the `store` command is not performed by the workflow action.
As I changed olny the last part of the workflow, instead of uploading the html report I've uploaded the report, I think that files and all other info are consistent.

**Comment by whisperity** [comment]
Oh, I understand it now... That repository is only meant as a throwaway repository to execute hand-written tests **in GitHub Actions** for the main Action code and it is not meant to be run *locally*. Documentation (and automated tests) about the action is available at [the GitHub Marketplace](http://github.com/marketplace/actions/codechecker-static-analysis). The action already supports uploading to a server automatically.

The defence I mentioned earlier is triggered based on file modification times, and if you download things between machines then it's understandable how those can get messed up.

You need to either:

1. Run the analysis in the cloud and do the store from the cloud, using [the appropriate configuration flags of the Action](http://github.com/marketplace/actions/codechecker-static-analysis#uploading-results-to-a-codechecker-server).
2. Run the analysis locally (you can find the steps for that in [CodeChecker's user guide](http://github.com/Ericsson/codechecker/blob/master/docs/usage.md)) and do the store from the computer where the analysis was run.

The `analyze` and `store` commands **must** run from the same computer, as the output of `analyze` that you give to `store` will have files that refer to other files that only exist in the context where `analyze` run!

(The only thing that can be done on another machine is after a `successful` store you can open the server in a Web browser, or via the `cmd` command, to **read** the bug reports.)

**Report-Converter** is only needed if you run analysis with a tool CodeChecker can't run natively, such as SpotBugs or PyLint. For C++ analyses with Clang, it is not needed.

**Comment by whisperity** [comment]
The problem is that, for you, `analyze` ran in the CI (on GitHub's cloud) and created a report structure that refers to `/home/runner/...`, which is a directory you likely do not have on your own computer &emdash; hence the warning _"source file ... missing"_.

**Comment by RobertoFioravanti** [comment]
thanks for the clarification
